docker_images/java/wrapper/src/main/java/glue/ModuleGlue.java

```diff
@@ -435,12 +435,12 @@ public void reset()
         }
 
         @Override
-        public DirectMethodResponse onMethodInvoked(String methodName, Object methodData, Object context)
+        public DirectMethodResponse onMethodInvoked(String methodName, DirectMethodPayload methodData, Object context)
         {
             System.out.printf("method %s called%n", methodName);
             if (methodName.equals(this._methodName))
             {
-                String methodDataString = new String((byte[]) methodData);
+                String methodDataString = methodData.getPayloadAsJsonString();
```

Comment:
Checking the value of `methodDataString` for passed tests, it seems like a json string which can still be deserialized.